src/components/NAlert/NAlert.vue

```diff
@@ -0,0 +1,408 @@
+            font-size: @BaseFontSize + 6;
+        }
+    }
+
+    &.n-alert-link-button-container {
+        .n-alert-label-text,
+        .n-alert-text {
+            font-size: @BaseFontSize + 4;
+        }
+    }
+
+    &.n-alert-info {
+        background: @ColorPrimary20;
+        border: 0.1rem solid @ColorPrimary50;
+
+        .n-alert-button {
+            background: @ColorPrimary50;
+        }
+    }
+
+    &.n-alert-success {
+        background: @Bubbles;
+        border: 0.1rem solid @SuccessColor;
+    }
+
+    &.n-alert-warn {
+        background: @ColorFeedbackWarning20;
+        border: 0.1rem solid @ColorFeedbackWarning50;
+    }
+
+    &.n-alert-error {
+        background: @ColorFeedbackError20;
+        border: 0.1rem solid @ColorFeedbackError50;
+    }
+
+    &.n-alert-loader-container {
+        background: @ColorPrimary20 !important;
+        border: 0.1rem solid @ColorPrimary50;
+    }
+
+    &.n-alert-custom-width {
+        .n-alert-content {
+            column-gap: 0.2rem;
+        }
+    }
+
+    &.n-alert-xs {
+        .n-alert-content {
+            flex-direction: column !important;
+            row-gap: 0.8rem;
+        }
+    }
+
+    .n-alert-content {
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        column-gap: 8rem;
+        width: 100%;
+        align-items: center;
+
+        &.n-alert-extended {
+            flex-direction: column;
+            align-items: unset;
+            justify-content: center;
+            row-gap: 1.2rem;
+
+            .n-alert-button-container {
+                display: flex;
+                justify-content: flex-end;
+            }
+        }
+    }
+
+    .n-alert-icon-text-wrapper {
+        display: flex;
+        align-items: center;
+        column-gap: 0.4rem;
+
+        .n-alert-icon {
+            height: 26.6px !important;
+            width: 26.6px !important;
+        }
+    }
+
+    .n-alert-loader {
+        height: 2.4rem;
+    }
+
+    .n-alert-button-container {
+        .n-alert-button {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            border-radius: 25rem;
+            padding: 0.4rem 1.2rem;
+            color: @WhiteColor;
+            font-weight: 700;
+            border: none;
+            cursor: pointer;
+
+            &:disabled {
+                cursor: not-allowed;
+                pointer-events: none;
+            }
+
+            &.n-alert-button-success {
+                background-color: @SuccessColor;
+            }
+
+            &.n-alert-button-warn {
+                background-color: @ColorFeedbackWarning50;
+            }
+
+            &.n-alert-button-error {
+                background-color: @ColorFeedbackError50;
+            }
+        }
+
+        .n-alert-link {
+            cursor: pointer;
+            font-weight: 700;
+            text-decoration: none;
+
+            &.n-alert-button-link-info {
```

Comment:
we can use class nesting instead of declaring class names fully

Reply:
The only problem here can be a massive loop. Again, I'll figure out next week.